pkg/kube/multicluster/secretcontroller.go

```diff
@@ -142,8 +136,12 @@ func NewController(kubeclientset kube.Client, namespace string, clusterID cluste
 	namespaces := kclient.New[*corev1.Namespace](kubeclientset)
 	controller.namespaces = namespaces
 	controller.DiscoveryNamespacesFilter = filter.NewDiscoveryNamespacesFilter(namespaces, meshWatcher.Mesh().GetDiscoverySelectors())
+	// Queue does NOT retry. The only error that can occur is if the kubeconfig is
+	// malformed. This is a static analysis that cannot be resolved by retry. Actual
+	// connectivity issues would result in HasSynced returning false rather than an
+	// error. In this case, things will be retried automatically (via informers or
+	// others), and the time is capped by RemoteClusterTimeout).
 	controller.queue = controllers.NewQueue("multicluster secret",
-		controllers.WithMaxAttempts(maxRetries),
```

Comment:
Do not remember the exact reason but I remember @hzxuzhonghu added this retry recently?

Reply:
It is because cluster adding/updating are complex, there are many calling stack, different errors can be returned

Reply:
And this pr is removing all the error return values, so look good